ImageAnalysis/image_analysis/base.py

```diff
@@ -81,3 +83,122 @@ def analyze_image(self,
 
         """
         raise NotImplementedError()
+
+
+class LabviewImageAnalyzer(ImageAnalyzer):
+    """
+    Intermediate class for ImageAnalyzer intended for analyzers that also want to be compatible with LabView through
+    labview_adapters.py.
+
+        Derived classes should implement
+            configure()
+            __init__()
+    """
+    def __init__(self, config_file=None, **kwargs):
+        """ Initializes the components of an ImageAnalyzer that talk with the functions in labview_adapters.py to create
+        an analyzer with settings defined in a config file and to allow for analyzer results to be passed back upwards
+        to Labview.
+
+        Parameters
+        ----------
+        config_file : str
+            Optional.  If given the analyzer is initialized using the settings in this config file location
+        kwargs :
+            Optional.  If no config file is given, one can explicitly name keyword arguments to overwrite during the
+            call to configure
+
+        """
+        super().__init__()
+        self.roi = None
+        self.background = None
+
+        if config_file:
+            self.apply_config(config_file)
+        else:
+            self.configure(**kwargs)
+
+    def apply_config(self, config_file):
+        """ Loads the config file and passes elements of 'settings' as keyword arguments to the configure function
+
+        Parameters
+        ----------
+        config_file : str
+            file location of the .ini config file
+
+        """
+        parser = configparser.ConfigParser()
+        parser.read(config_file)
+        if 'roi' in parser:
+            self.roi = self.read_roi(parser)
+        config = dict(parser["settings"])
+        self.configure(**config)
+
+    @staticmethod
+    def read_roi(parser):
+        """ Reads the roi settings from the .ini config file
+
+        Parameters
+        ----------
+        parser : ConfigParser
+            the config file containing roi information
+
+        Returns
+        -------
+        roi : 1d array
+            the roi bounds given by [top, bottom, left, right] pixel
+
+        """
+        roi_top = int(parser.get('roi', 'top')),
+        roi_bottom = int(parser.get('roi', 'bottom')),
+        roi_left = int(parser.get('roi', 'left')),
+        roi_right = int(parser.get('roi', 'right')),
+        return np.array([roi_top, roi_bottom, roi_left, roi_right]).reshape(-1)
+
+    def roi_image(self, image):
```

Comment:
Note that if self.roi is of [ROI](https://github.com/GEECS-BELLA/GEECS-Plugins/blob/8ca9adaa8dace69f78d0735386a17b93bdfb9d28/ImageAnalysis/image_analysis/utils.py#L48 )type, it has a .crop method.

Reply:
Right, we start using that if ROI works on the Labview python version.  I'll try to test and debug that in the near future.